src/Data/OpenApi/Internal.hs

```diff
@@ -838,8 +838,17 @@ data OAuth2Flows = OAuth2Flows
   , _oAuth2FlowsAuthorizationCode :: Maybe (OAuth2Flow OAuth2AuthorizationCodeFlow)
   } deriving (Eq, Show, Generic, Data, Typeable)
 
+data HttpSchemeType
+  = HttpSchemeBearer
+  | HttpSchemeBasic
```

Comment:
The spec (https://swagger.io/specification/#security-scheme-object) for `scheme` property with `http` type says:

> The values used SHOULD be registered in the [IANA Authentication Scheme registry](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml).

So I'd say we should either add other options as well (`Digest`, `HOBA`, ...) or at least provide `HttpSchemeCustom Text` constructor.

Also, for `Bearer` scheme the spec mentions an optional field `bearerFormat`:

> A hint to the client to identify how the bearer token is formatted. Bearer tokens are usually generated by an authorization server, so this information is primarily for documentation purposes.

Please add it to the `HttpSchemeBearer` constructor as `Maybe Text` field.